tests/ui/half-open-range-patterns/range_pat_interactions2.rs

```diff
@@ -8,8 +8,7 @@ fn main() {
     for x in -9 + 1..=(9 - 2) {
         match x as i32 {
             0..=(5+1) => errors_only.push(x),
-            //~^ error: inclusive range with no end
-            //~| error: expected one of `=>`, `if`, or `|`, found `(`
+            //~^ error: expected `)`, found `+`
```

Comment:
Ah, this is not ideal but you probably plan on fixing this in a follow-up PR or in #118625 (I haven't looked at the changes in that PR)?

We could create parser snapshots and try to parse a *RangePatternBound* first, then a *Pattern* on failure and finally an *Expression* on failure and provide targeted diagnostics accordingly.

Reply:
I changed it to say « unexpected `(` » if *RangePatternBound* didn't parse or if it did but didn't end with `)` , but yeah #118625 will try to parse an *Expression* to emit a better diagnostic:
```
error: expected pattern bound, found expression
  --> $DIR/range_pat_interactions2.rs:10:18
   |
LL |             0..=(5+1) => errors_only.push(x),
   |                  ^^^ expressions are not allowed in patterns

error: range pattern bounds cannot have parentheses
  --> $DIR/range_pat_interactions2.rs:10:17
   |
LL |             0..=(5+1) => errors_only.push(x),
   |                 ^   ^
   |
help: remove these parentheses
   |
LL -             0..=(5+1) => errors_only.push(x),
LL +             0..=5+1 => errors_only.push(x),
   |
   ```
However #118625 got a bit too big, so I split it into sub-PRs and I'll rebase it once this one has been merged.